pkg/server/singleprocess/service.go

```diff
@@ -141,16 +141,18 @@ func New(opts ...Option) (pb.WaypointServer, error) {
 		s.id = id
 	}
 
-	// If we haven't initialized our server config before, do that once.
-	conf, err := state.ServerConfigGet()
-	if err != nil {
-		return nil, err
-	}
-	if conf.Cookie == "" {
-		err := state.ServerConfigSet(conf)
-		if err != nil && status.Convert(err).Code() != codes.Unimplemented {
+	if !cfg.skipServerConfigInit {
```

Comment:
This is a mostly unrelated change - it allows the server to be started without initializing a global cookie.